sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -559,7 +563,10 @@ public void dispose() {
             }
 
             if (WebExceptionUtility.isNetworkFailure(dce)) {
-                BridgeInternal.setSubStatusCode(dce, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_UNAVAILABLE);
+                if (!(dce.getCause() instanceof ReadTimeoutException)) {
```

Comment:
Will it impact the full region fail-over?
If so does it needs to be finer scoped like, ReadTimeout & non-data-plane operations etc...

Reply:
Since it doesn't impacts the Region failover, resolving this conversation.